app/uk/gov/hmrc/vatsubscription/controllers/actions/VatAuthorised.scala

```diff
@@ -41,13 +45,17 @@ class VatAuthorised @Inject()(val authConnector: AuthConnector, implicit val app
 
   def async(vrn: String)(f: User[_] => Future[Result])(implicit ec: ExecutionContext): Action[AnyContent] = Action.async {
     implicit request =>
-      authorised(delegatedAuthRule(vrn)).retrieve(Retrievals.allEnrolments and Retrievals.credentials) {
-        case enrolments ~ credentials =>
+      authorised(delegatedAuthRule(vrn)).retrieve(retrieve.allEnrolments and retrieve.credentials) {
+        case  enrolments ~ Some(credentials) =>
           f(User(vrn, arn(enrolments), credentials.providerId)(request))
+        case _ =>
+          Logger.warn(s"[VatAuthorised][async] - Unable to retrieve credentials/None in field")
```

Comment:
```suggestion
          Logger.warn(s"[VatAuthorised][async] - Unable to retrieve Credentials.providerId from auth profile")
```
I think this message is slightly more informative